### ISSUE TITLE
The list instance's "FieldDefaults" element is not applied to documents that are uploaded as part of a folder

### ISSUE THREAD

**Issue by LarterStyles** [SharePoint/PnP-Provisioning-Schema#374]
When specifying a **taxonomy column's default value** at the **list level** (via the FieldDefault property, see below), the metadata is populated correctly when only a **single document** is dragged 'n dropped into the library. When a folder containing one or more documents is uploaded however, the fields aren't set on the documents at all.

```
<pnp:ListInstance Title="Human Resources"
                  Url="HR"
                  TemplateType="101"
                  TemplateFeatureID="00bfea71-e717-4e80-aa17-d0c71b360101"
                  ListExperience="NewExperience"
                  ContentTypesEnabled="true"
                  EnableAttachments="true"
                  EnableFolderCreation="true"
                  EnableModeration="true"
                  EnableVersioning="true"
                  EnableMinorVersions="true"
                  DraftVersionVisibility="0"
                  IsApplicationList="false">
          <pnp:ContentTypeBindings>
            ...
          </pnp:ContentTypeBindings>
          <pnp:FieldDefaults>
            <pnp:FieldDefault FieldName="Department">-1;#Human Resources|26f8ffc9-bd94-4516-b3f4-a1bb9c7561a6</pnp:FieldDefault>
          </pnp:FieldDefaults>
```

I've repro'd this in multiple libraries, in multiple tenants. O365 support have also repro'd it on their end.

Funnily enough, if you click on the list column in the library settings page (where should now see that the column has a default value just for this library), then click OK **without making any changes at all**, then everything works as expected. This suggests that PnP is doing something different to the OOTB SP UI.

And finally, if you then redeploy the PnP schema posted above, it reverts back to the "broken" behaviour, so we eseentially have to comment out the `<pnp:FieldDefaults>` block altogether.

**Comment by LarterStyles** [comment]
An update on this: the MS Support team has confirmed that default colum values, when applied at the **site column or the library level**, are set based on the item’s content type. In other words, when you upload a single document, it only gets the default value if it it’s based on a content type that has that specific column added to it. No surprises here.

The problem described above however, is occuring because when I’m uploading files “in bulk”, I was actually dragging a folder into the library (which contains 100+ documents). Well, since the folder content type doesn’t have any of my special columns added to it, it rightfully doesn’t get populated with any default values — but for whatever reason all of it’s child items are treated the same way / skipped.

If you set the default at the **folder level** though, it doesn’t care about content types, and so you’ll see your default value appearing on folders and any other kinds of items that you upload. In my case, this is undesirable behaviour.

So in short, if you’re setting default values via PnP on the list instance element, you better disable folder creation, otherwise you’re going to missing metadata...

**Comment by PaoloPia** [comment]
Hi,
Thanks for your feedback. This is an engine related issue/request. Could you please insert this item, if still needed, in the PnP-Sites-Core repository (https://github.com/SharePoint/PnP-Sites-Core/issues/)? Thanks. I will close this item, because out of context.
